src/Reflection/Adapter/ReflectionClass.php

```diff
@@ -68,6 +70,22 @@ public function __toString()
         return $this->betterReflectionClass->__toString();
     }
 
+    /**
+     * @param string $name
+     *
+     * @return mixed
+     *
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
+     */
+    public function __get($name)
+    {
+        if ($name === 'name') {
+            return $this->betterReflectionClass->getName();
+        }
+
+        return $this->betterReflectionClass->{$name};
```

Comment:
Can we test this path, or maybe throw an exception?

Reply:
Yep, throwing an exception seems more sensible here: we just need a test case for it :)

Reply:
I'd be more comfortable with 1:1 behaviour with PHP's reflection class :)

Reply:
In which case you'd need a nightmare like https://github.com/Ocramius/ProxyManager/blob/master/src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php :P 

Ok, we just need a test raising an error then (and asserting on it)

Reply:
Oh god. Alright, what should be thrown?

Reply:
an `OutOfBoundsException`, probably

Reply:
Updated.